src/hotspot/share/opto/callGenerator.cpp

```diff
@@ -731,7 +731,7 @@ void CallGenerator::do_late_inline_helper() {
       C->env()->notice_inlined_method(inline_cg()->method());
     }
     C->set_inlining_progress(true);
-    C->set_do_cleanup(kit.stopped()); // path is dead; needs cleanup
+    C->set_do_cleanup(kit.stopped() || result->Opcode() == Op_VectorBox); // path is dead or vector box; needs cleanup
```

Comment:
This only triggers if the return value of the incrementally inlined method is a `VectorBox`, right? Is that sufficient? Could the `VectorBox` be hidden by another node?

Reply:
I'm failing to understand why this is only an issue with VectorBox.  It doesn't feel quite right to be checking for a specific node type here.  Maybe this should be something like needs_cleanup(result)?